rclrs/src/executor.rs

```diff
@@ -1,48 +1,58 @@
-use crate::{rcl_bindings::rcl_context_is_valid, Node, RclReturnCode, RclrsError, WaitSet};
+use crate::{
+    rcl_bindings::rcl_context_is_valid, ContextHandle, Node, NodeOptions, RclrsError, WaitSet,
+    WeakNode,
+};
 use std::{
-    sync::{Arc, Mutex, Weak},
+    sync::{Arc, Mutex},
     time::Duration,
 };
 
 /// Single-threaded executor implementation.
-pub struct SingleThreadedExecutor {
-    nodes_mtx: Mutex<Vec<Weak<Node>>>,
+pub struct Executor {
```

Comment:
> I wonder if by renaming a SingleThreadedExecutor to just Executor it might be a bit confusing if in the future we want to add a new type of executor.

In #421 this `Executor` class will become an opaque wrapper for a user-selected executor. That means the user will be able to choose, at runtime, any kind of executor they want (single-threaded, multi-threaded, custom implementation, whatever) and the rest of the rclrs code will still just see it as the same `Executor`.

`Context::create_basic_executor` is used in #421 to create the single-threaded async executor that we provide out-of-the-box. By introducing `Context::create_basic_executor` it in this PR, users will not experience any API breakage when #421 is merged.